python/pyspark/mllib/clustering.py

```diff
@@ -86,6 +86,68 @@ def train(cls, rdd, k, maxIterations=100, runs=1, initializationMode="k-means||"
         return KMeansModel([c.toArray() for c in centers])
 
 
+class GaussianMixtureModel(object):
+
+    """A clustering model derived from the Gaussian Mixture Model method.
+
+    >>> from numpy import array
+    >>> clusterdata_1 =  sc.parallelize(array([-0.1,-0.05,-0.01,-0.1,
+    ...                                         0.9,0.8,0.75,0.935,
+    ...                                        -0.83,-0.68,-0.91,-0.76 ]).reshape(6,2))
+    >>> model = GaussianMixtureEM.train(clusterdata_1, 3, 0.0001, 3205, 10)
+    >>> labels = model.predictLabels(clusterdata_1).collect()
+    >>> labels[0]==labels[2]
+    True
+    >>> labels[3]==labels[4]
+    False
+    >>> labels[4]==labels[5]
+    True
+    >>> clusterdata_2 =  sc.parallelize(array([-5.1971, -2.5359, -3.8220,
+    ...                                        -5.2211, -5.0602,  4.7118,
+    ...                                         6.8989, 3.4592,  4.6322,
+    ...                                         5.7048,  4.6567, 5.5026,
+    ...                                         4.5605,  5.2043,  6.2734]).reshape(5,3))
+    >>> model = GaussianMixtureEM.train(clusterdata_2, 2, 0.0001, 150, 10)
+    >>> labels = model.predictLabels(clusterdata_2).collect()
+    >>> labels[0]==labels[1]==labels[2]
+    True
+    >>> labels[3]==labels[4]
+    True
+    """
+
+    def __init__(self, weight, mu, sigma):
```

Comment:
FYI, the names were changed to `weights` and `gaussians`. We might want to discuss how the Python API should change.